core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -66,14 +66,19 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
     }
   }
 
-  // Print a warning if the user did not select the APU architecture when using
-  // a MI300A
+  // Print a warning if the user did not select the right GFX942 architecture
 #ifdef KOKKOS_ARCH_AMD_GFX942
   if ((Kokkos::show_warnings()) && (hipProp.integrated == 1)) {
     std::cerr << "Kokkos::HIP::initialize WARNING: running kernels for MI300X "
                  "(discrete GPU) on a MI300A (APU).\n";
   }
 #endif
+#ifdef KOKKOS_ARCH_AMD_GFX942_APU
+  if ((Kokkos::show_warnings()) && (hipProp.integrated == 0)) {
+    std::cerr << "Kokkos::HIP::initialize WARNING: running kernels for MI300A "
+                 "(APU) on a MI300X (discrete GPU).\n";
+  }
```

Comment:
I get an error when enabling `KOKKOS_ARCH_AMD_GFX942_APU` with rocm 6.1:
```
kokkos/core/src/HIP/Kokkos_HIP.cpp:77:37: error: use of undeclared identifier 'hipProp'
   77 |   if ((Kokkos::show_warnings()) && (hipProp.integrated == 0)) {
      |   
```

Should there be something like this:
```
hipDeviceProp_t hipProp;
hipGetDeviceProperties(&hipProp, 0);
```

Reply:
It should be fixed. I should have use `Impl::HIPInternal::m_deviceProp`